src/Payments/PaymentAmount.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Webit\WFirmaSDK\Payments;
+
+final class PaymentAmount
+{
+    private const ACCOUNT_PLN = 'pln';
+    private const ACCOUNT_CURRENCY = 'currency';
+
+    /** @var float */
+    private $value;
+
+    /** @var string|null */
+    private $account;
+
+    /** @var float|null */
+    private $valuePln;
+
+    /** @var float|null */
+    private $exchangeRate;
+
+    /** @var string|null */
+    private $nbpLabel;
+
+    /** @var \DateTime|null */
+    private $rateDate;
+
+    /**
+     * @internal To create this object use named constructors `forPlnAccount` or `forCurrencyAccount`
+     *
+     * @param float $value
+     * @param ?string $account
+     * @param float|null $valuePln
+     * @param float|null $exchangeRate
+     * @param string|null $nbpLabel
+     * @param \DateTime|null $rateDate
+     */
+    public function __construct(float $value, ?string $account = null, ?float $valuePln = null, ?float $exchangeRate = null, ?string $nbpLabel = null, \DateTime $rateDate = null)
+    {
+        $this->value = $value;
+        $this->account = $account;
+        $this->valuePln = $valuePln;
+        $this->exchangeRate = $exchangeRate;
+        $this->nbpLabel = $nbpLabel;
+        $this->rateDate = $rateDate;
+    }
+
+    /**
+     * @return ?float
+     */
+    public function value(): ?float
+    {
+        return $this->value;
+    }
+
+    /**
+     * @return ?string
+     */
+    public function account(): ?string
+    {
+        return $this->account;
+    }
+
+    /**
+     * @return ?float
+     */
+    public function valuePln(): ?float
+    {
+        return $this->valuePln;
+    }
+
+    /**
+     * @return ?float
+     */
+    public function exchangeRate(): ?float
+    {
+        return $this->exchangeRate;
+    }
+
+    /**
+     * @return ?string
+     */
+    public function nbpLabel(): ?string
+    {
+        return $this->nbpLabel;
+    }
+
+    /**
+     * @return ?\DateTime
+     */
+    public function rateDate(): ?\DateTime
+    {
+        return $this->rateDate;
+    }
+
+    /**
+     * @param float
+     * @param float|null $valuePln
+     * @return PaymentAmount
+     */
+    public static function forPlnAccount(float $value, ?float $valuePln = null): PaymentAmount
+    {
+        return new self($value, self::ACCOUNT_PLN, $valuePln);
+    }
+
+    /**
+     * @param float
+     * @param float|null $valuePln
+     * @return PaymentAmount
+     */
+    public static function forCurrencyAccount(float $value, ?float $valuePln = null): PaymentAmount
```

Comment:
Tutaj "valuePln" nie ma sensu - jak płacimy w walucie, to nie interesuje nas ile zapłaciliśmy w złotówkach (system wyliczy nam kurs księgowy samodzielnie - pola currency_*)

Reply:
Currency zachowuje się różnie w zależności czy faktura jest wystawiona z PLN czy w innej walucie. Bawiłem się tym troszkę, i jest jakiś przypadek, że można wpisać z palca valuePln (chyba w sytuacji, jak faktura jest w PLN, a wpłata jest na currencyAccount, wtedy możesz podać wartość wpłaty + wartość wpłaty w PLN, nie ma automatycznej konwersji w takiej sytuacji). Nie wiem, czy takie coś występuje w naturze, ale API to łyka.